notebook/static/notebook/js/notebook.js

```diff
@@ -1380,20 +1380,23 @@ define(function (require) {
      * Replace the selected cell with the cells in the clipboard.
      */
     Notebook.prototype.paste_cell_replace = function () {
-        if (this.clipboard !== null && this.paste_enabled) {
-            var first_inserted = null;
-            for (var i=0; i < this.clipboard.length; i++) {
-                var cell_data = this.clipboard;
-                var new_cell = this.insert_cell_above(cell_data.cell_type);
-                new_cell.fromJSON(cell_data);
-                if (first_inserted === null) {
-                    first_inserted = new_cell;
-                }
-            }
-            var old_selected = this.get_selected_index();
-            this.select(this.find_cell_index(first_inserted));
-            this.delete_cell(old_selected);
+
+        if (!(this.clipboard !== null && this.paste_enabled)) {
+            return
+        }
+
+        var selected =  this.get_selected_cells_indices();
+        var insertion_index = selected[0];
```

Comment:
Assumes `get_selected_cells_indices` is least to greatest, is this an okay assumption?  Else `Math.min.apply(Math, selected)`